src/test/java/suite/optout/OptoutTest.java

```diff
@@ -151,10 +170,56 @@ public void testV2TokenRefreshAfterOptOut(String label, Operator operator, Strin
         with().pollInterval(5, TimeUnit.SECONDS).await("Get V2 Token Response").atMost(OPTOUT_WAIT_SECONDS, TimeUnit.SECONDS).until(() -> operator.v2TokenRefresh(refreshToken, refreshResponseKey).equals(Mapper.OBJECT_MAPPER.readTree("{\"status\":\"optout\"}")));
     }
 
+    @Order(9)
+    @ParameterizedTest(name = "/v2/optout/status after v2/identity/map and v2/token/logout - DII {0} - expecting {4} - {2}")
+    @MethodSource({"afterOptoutAdvertisingIdArgs"})
+    public void testV2OptOutStatus(String label, Operator operator, String operatorName, String rawUID,
+                                   boolean isOptedOut, long optedOutTimestamp) throws Exception {
+        String payload = "{\"advertising_ids\":[\"" + rawUID + "\"]}";
+        with().pollInterval(5, TimeUnit.SECONDS)
+                .await("Get Opt out status response")
+                .atMost(OPTOUT_WAIT_SECONDS, TimeUnit.SECONDS)
+                .until(() -> {
+                    JsonNode response = operator.v2OptOutStatus(payload);
+                    JsonNode body = response.get("body");
+                    JsonNode optedOutJsonNode = body.get("opted_out");
+                    if (isOptedOut) {
+                        return optedOutJsonNode.size() == 1;
+
+                    } else {
+                        return optedOutJsonNode.isEmpty();
+                    }
+                });
+        JsonNode response = operator.v2OptOutStatus(payload);
+        assertThat(response.at("/status").asText()).isEqualTo("success");
+        JsonNode body = response.get("body");
+        assertThat(body).isNotNull();
+        JsonNode optedOutJsonNode = body.get("opted_out");
+        assertThat(optedOutJsonNode).isNotNull();
+        if (isOptedOut) {
+            assertThat(optedOutJsonNode.size()).isEqualTo(1);
+            JsonNode optedOutRecord = optedOutJsonNode.get(0);
+            assertThat(optedOutRecord).isNotNull();
+            assertThat(optedOutRecord.has(TestData.ADVERTISING_ID)).isTrue();
+            String advertisingId = optedOutRecord.get(TestData.ADVERTISING_ID).asText();
+            assertThat(advertisingId).isEqualTo(rawUID);
+            assertThat(optedOutRecord.has(TestData.OPTED_OUT_SINCE)).isTrue();
+            long optedOutSince = optedOutRecord.get(TestData.OPTED_OUT_SINCE).asLong();
+            // TODO uncomment after Opt Out Status API returns timestamp in milliseconds
+//             assertThat(optedOutSince).isGreaterThanOrEqualTo(optedOutTimestamp);
```

Comment:
Do you wanna fix the timestamp first before merging this? Or just treat the timestamp as seconds and do optedOutSince*1000 (i think) in this MR and merge it first and once you fix the timestamp you fix this test again?

Reply:
Ok, I'll fix the timestamp first and then merge this. Less likely this code changes in the next few days and cause merge conflicts. Will save me additional work to come back and fix the code